File: programmers/코딩테스트 고득점 Kit/해시/위장.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;
map<string, int> m;
map<string, int> chk;
int solution(vector<vector<string>> clothes) {
	int answer = 1;
	int cnt = 0;
	for (int i = 0; i < clothes.size(); i++)
		m[clothes[i][1]]++;
	for (int i = 0; i < clothes.size(); i++) {
		if (chk[clothes[i][1]] == 0) {
			chk[clothes[i][1]] = 1;
			answer *= (m[clothes[i][1]] + 1);
		}
	}
	return answer - 1;
}

File: programmers/코딩테스트 고득점 Kit/해시/위장.java
import java.util.*;
class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, Integer> m = new HashMap<>();
        for(int i=0;i<clothes.length;i++) {
            String type = clothes[i][1];
            if(m.containsKey(type)) m.put(type, m.get(type) + 1);
            else  m.put(type, 1);
        }
        for(int i=0;i < clothes.length;i++) {
            String type = clothes[i][1];
            if(m.get(type) != 0) {
                answer *= m.get(type) + 1;
                m.put(type, 0);
            }
        }
        return answer - 1;
    }
}